packages/minimongo/selector.js

```diff
@@ -649,6 +690,108 @@ ELEMENT_OPERATORS = {
+
+        return typeof _.find(operand, function (op, i) {
+            var bitsSetOp = get8BitsSet(op);
+            var bitsSetVal = get8BitsSet(value[i]);
+
+            return typeof _.find(bitsSetOp, function (bit) {
+                return bitsSetVal.indexOf(bit) === -1;
+              }) === 'undefined';
+          }) === 'undefined';
+      };
+    }
+  },
+  $bitsAllClear: {
+    compileElementSelector: function (operand, valueSelector, matcher) {
+      if (!isArray(operand) && !(operand instanceof Uint8Array))
+        throw Error("$bitsAllClear has to be an Array");
+      return function (value) {
+        if (typeof value !== 'number' && !(value instanceof Uint8Array))
+          return false;
+
+        if (value === 0)
+          return true;
+
+        operand = ensureOperandUint8Array(operand);
+
+        if (!(value instanceof Uint8Array))
+          value = convertNumberToUint8Array(value);
+
+        return _.filter(operand, function (op, i) {
+          var bitsSetOp = get8BitsSet(op);
+          var bitsSetVal = get8BitsSet(value[i]);
+
+          return typeof _.find(bitsSetOp, function (bit) {
+            return bitsSetVal.indexOf(bit) !== -1;
+          }) !== 'undefined';
+        }).length === 0;
+      };
+    }
+  },
+  $bitsAllSet: {
+    compileElementSelector: function (operand, valueSelector, matcher) {
+      if (!isArray(operand) && !(operand instanceof Uint8Array))
+        throw Error("$bitsAllSet has to be an Array");
+      return function (value) {
+        if (typeof value !== 'number' && !(value instanceof Uint8Array))
+          return false;
+
+        operand = ensureOperandUint8Array(operand);
+
+        if (!(value instanceof Uint8Array))
+          value = convertNumberToUint8Array(value);
+
+        return _.filter(operand, function (op, i) {
+            var bitsSetOp = get8BitsSet(op);
+            var bitsSetVal = get8BitsSet(value[i]);
+
+            return typeof _.find(bitsSetOp, function (bit) {
+                return bitsSetVal.indexOf(bit) === -1;
+              }) !== 'undefined';
+          }).length === 0;
+      };
+    }
+  },
+  $bitsAnySet: {
+    compileElementSelector: function (operand, valueSelector, matcher) {
+      if (!isArray(operand) && !(operand instanceof Uint8Array))
+        throw Error("$bitsAnySet has to be an Array");
+      return function (value) {
+        if (typeof value !== 'number' && !(value instanceof Uint8Array))
+          return false;
+
+        operand = ensureOperandUint8Array(operand);
+
+        if (!(value instanceof Uint8Array))
+          value = convertNumberToUint8Array(value);
+
+        return typeof _.find(operand, function (op, i) {
+            var bitsSetOp = get8BitsSet(op);
+            var bitsSetVal = get8BitsSet(value[i]);
+
+            return typeof _.find(bitsSetOp, function (bit) {
+                return bitsSetVal.indexOf(bit) !== -1;
+              }) === 'undefined';
+          }) === 'undefined';
+      };
+    }
+  },
```

Comment:
There seems to be a fair bit of duplicated code between these methods. Can you factor some of that out into helper functions? For example, this expression is repeated verbatim:
```js
_.filter(operand, function (op, i) {
  var bitsSetOp = get8BitsSet(op);
  var bitsSetVal = get8BitsSet(value[i]);

  return typeof _.find(bitsSetOp, function (bit) {
    return bitsSetVal.indexOf(bit) !== -1;
  }) !== 'undefined';
})
```